src/plugins/intel_gpu/include/intel_gpu/primitives/matrix_nms.hpp

```diff
@@ -0,0 +1,169 @@
+        // specifies whether boxes are normalized or not
+        bool normalized = true;
+
+        attributes() {}
+
+        attributes(const ngraph::op::v8::MatrixNms::Attributes& attrs)
+            : attributes(from(attrs.sort_result_type),
+                         attrs.sort_result_across_batch,
+                         attrs.score_threshold,
+                         attrs.nms_top_k,
+                         attrs.keep_top_k,
+                         attrs.background_class,
+                         from(attrs.decay_function),
+                         attrs.gaussian_sigma,
+                         attrs.post_threshold,
+                         attrs.normalized) {}
+
+        attributes(SortResultType sort_result_type,
+                   bool sort_result_across_batch,
+                   float score_threshold,
+                   int nms_top_k,
+                   int keep_top_k,
+                   int background_class,
+                   DecayFunction decay_function,
+                   float gaussian_sigma,
+                   float post_threshold,
+                   bool normalized)
+            : sort_result_type(sort_result_type),
+              sort_result_across_batch(sort_result_across_batch),
+              score_threshold(score_threshold),
+              nms_top_k(nms_top_k),
+              keep_top_k(keep_top_k),
+              background_class(background_class),
+              decay_function(decay_function),
+              gaussian_sigma(gaussian_sigma),
+              post_threshold(post_threshold),
+              normalized(normalized) {}
+    };
+
+    /// @brief Constructs matrix_nms primitive.
+    /// @param id This primitive id.
+    /// @param boxes primitive id.
+    /// @param scores primitive id.
+    /// @param second_output primitive id.
+    /// @param third_output primitive id.
+    /// @param attrs attributes.
+    matrix_nms(const primitive_id& id,
+               const primitive_id& boxes,
+               const primitive_id& scores,
+               const primitive_id& second_output,
+               const primitive_id& third_output,
+               const matrix_nms::attributes& attrs)
+        : primitive_base(id, {boxes, scores}),
+          second_output(second_output),
+          third_output(third_output),
+          attribs(attrs) {}
+
+    /// @brief Constructs matrix_nms primitive.
+    /// @param id This primitive id.
+    /// @param boxes primitive id.
+    /// @param scores primitive id.
+    /// @param second_output primitive id.
+    /// @param third_output primitive id.
+    /// @param attrs operation attributes.
+    matrix_nms(const primitive_id& id,
+               const primitive_id& boxes,
+               const primitive_id& scores,
+               const primitive_id& second_output,
+               const primitive_id& third_output,
+               const ngraph::op::v8::MatrixNms::Attributes& attrs)
+        : primitive_base(id, {boxes, scores}),
```

Comment:
if the second and third output is mandatory maybe it make sense to put it here to parent?

Reply:
maybe response to not this comment, here I was about using
`primitive_base(id, {boxes, scores, second_output, third_output})`

Reply:
agree, we use get_dependencies mechanism for conditional inputs only